pandas/core/frame.py

```diff
@@ -3489,33 +3479,34 @@ def _get_info_slice(obj, indexer):
                 )
             )
 
-        # empty include/exclude -> defaults to True
-        # three cases (we've already raised if both are empty)
-        # case 1: empty include, nonempty exclude
-        # we have True, True, ... True for include, same for exclude
-        # in the loop below we get the excluded
-        # and when we call '&' below we get only the excluded
-        # case 2: nonempty include, empty exclude
-        # same as case 1, but with include
-        # case 3: both nonempty
-        # the "union" of the logic of case 1 and case 2:
-        # we get the included and excluded, and return their logical and
-        include_these = Series(not bool(include), index=self.columns)
-        exclude_these = Series(not bool(exclude), index=self.columns)
-
-        def is_dtype_instance_mapper(idx, dtype):
-            return idx, functools.partial(issubclass, dtype.type)
-
-        for idx, f in itertools.starmap(
-            is_dtype_instance_mapper, enumerate(self.dtypes)
-        ):
-            if include:  # checks for the case of empty include or exclude
-                include_these.iloc[idx] = any(map(f, include))
-            if exclude:
-                exclude_these.iloc[idx] = not any(map(f, exclude))
+        # We raise when both include and exclude are empty
+        # Hence, we can just shrink the columns we want to keep
+        keep_these = np.full(self.shape[1], True)
+
+        def extract_unique_dtypes_from_dtypes_list(
+            dtypes_list: List[Type], unique_dtypes: List[Type]
```

Comment:
```suggestion
            dtypes_list: FrozenSet[Dtype], unique_dtypes: np.ndarray
```

Looking at code I don't think either of the arguments here are lists. You can import `Dtype` from pandas._typing and use that to subscript the first argument. The latter appears to be an ndarray, which unfortunately doesn't support subscripting